applications/GeoMechanicsApplication/tests/test_t_normal_flux_condition.cpp

```diff
@@ -0,0 +1,260 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Gennady Markelov
+//
+
+#include "containers/model.h"
+#include "custom_conditions/T_normal_flux_condition.h"
+#include "geo_mechanics_application_variables.h"
+#include "testing/testing.h"
+
+using namespace Kratos;
+
+namespace Kratos::Testing {
+
+void TestTnormalFluxCondition(ModelPart& rModelPart)
+{
+    Condition::Pointer p_condition = rModelPart.pGetCondition(1);
+
+    for (unsigned int i = 0; i < rModelPart.NumberOfNodes(); i++) {
+        p_condition->GetGeometry()[i].AddDof(TEMPERATURE);
+    }
+
+    const ProcessInfo& r_current_process_info = rModelPart.GetProcessInfo();
+    // p_condition->Check(r_current_process_info);
+    // Check requires a solution? 
+
+    Condition::DofsVectorType dof_list;
+    p_condition->GetDofList(dof_list, r_current_process_info);
+
+    for (unsigned int i = 0; i < dof_list.size(); i++) {
+        dof_list[i]->SetEquationId(i);
+    }
+
+    Condition::EquationIdVectorType equation_id_vector;
+    p_condition->EquationIdVector(equation_id_vector, r_current_process_info);
+
+    // Check the EquationIdVector values
+    for (unsigned int i = 0; i < equation_id_vector.size(); i++) {
+        KRATOS_EXPECT_TRUE(equation_id_vector[i] == i);
+    }
+}
+
+void GenerateTnormalFluxCondition2D2N(ModelPart& rModelPart)
+{
+	rModelPart.GetProcessInfo()[DOMAIN_SIZE] = 2;
+
+	rModelPart.CreateNewProperties(0);
+	Properties::Pointer p_cond_prop = rModelPart.pGetProperties(0);
```

Comment:
CreateNewProperties also directly returns the pointer, so this could be simplified to:
```suggestion
	Properties::Pointer p_cond_prop = rModelPart.CreateNewProperties(0);
```

Reply:
Thank you. It is better.